forte/data/readers/base_reader.py

```diff
@@ -270,26 +237,24 @@ def cache_data(self,
         logger.info("Caching pack to %s", cache_filename)
         if append:
             with open(cache_filename, 'a') as cache:
-                cache.write(self.serialize_instance(pack) + "\n")
+                cache.write(pack.serialize() + "\n")
         else:
             with open(cache_filename, 'w') as cache:
-                cache.write(self.serialize_instance(pack) + "\n")
+                cache.write(pack.serialize() + "\n")
 
     def read_from_cache(self, cache_filename: Path) -> Iterator[PackType]:
-        """
-        Reads one or more Packs from a cache_filename,
-        and yields Pack(s) from the cache file.
+        r"""Reads one or more Packs from ``cache_filename``, and yields Pack(s)
+        from the cache file.
 
         Args:
-            cache_filename: Path to the cache file
+            cache_filename: Path to the cache file.
 
         Returns: List of cached data packs.
-
         """
         logger.info("reading from cache file %s", cache_filename)
         with cache_filename.open("r") as cache_file:
             for line in cache_file:
-                pack = self.deserialize_instance(line.strip())
+                pack = jsonpickle.decode(line.strip())
```

Comment:
I am wondering instead of `jsonpickle.decode(...)` we can provide a class method inside `MultiPack` and `DataPack` for deserialize. The reason is - if we serialize packs using `pack.serialize(...)` we would like to deserialize it using `DataPack.deserialize(...)` or `MultiPack.deserialize(...)` because the serialization and deserialization should be handled by the pack.